### DISCUSSION TITLE
Debug

### DISCUSSION THREAD

**Discussion by contatolucascosta** [schibsted/jslt#169] [Ideas]
Hi, is there a way to implement and use Debug Mode for JSLT Templates ? Technically is it possible ?

I believe that would be amazing and easier for developers. Even if is needs to use a specific IDE or maybe a new Playground to cover this possible feature.

**Comment by larsga** [comment]
What kind of debugging features were you thinking of?

**Comment by catull** [comment]
When I developed my custom functions, I always added extra attributes such as `temp` into the mix, in order to verify temporary variables had the correct value.

This worked great in recursive function calls.

Once I had the code working, I dropped the extra `temp` attributes.

If this is not your use case, please describe the problem you are facing.

**Comment by contatolucascosta** [comment]
I was wondering if we had debug for each line code. it would be util for complex codes, for example using chain instructions. If would possible debug line by line and function by function and concepts of breakpoint, something like debug in Java and Eclipse IDE (eg.)
